workspace/__lib__/xod/waves/variable-triangle-wave-map/patch.cpp

```diff
@@ -0,0 +1,17 @@
+
+struct State {
+};
+
+{{ GENERATED_CODE }}
+
+void evaluate(Context ctx) {
+    auto fractionalPart = fmod(getValue<input_X>(ctx), 1);
+    auto peakPosition = constrain(getValue<input_PP>(ctx), 0.0, 1.0);
```

Comment:
I got an error in simulation:
```
error: use of undeclared identifier 'constrain'
    auto peakPosition = constrain(getValue<input_PP>(ctx), 0.0, 1.0);
                        ^
1 error generated.
```

So you probably have to use conditions or make a patch to our wasm package with the implementation of this function (and pray to God that every Arduino package developers add it into `core.h` 😄 ).

Reply:
> and pray to God that every Arduino package developers add it into core.h 😄

I'm pretty sure they do. It's not like it's some obscure undocumented function: https://www.arduino.cc/reference/en/#functions